keps/sig-auth/4193-tokenrequest-requester-info-enhancements/README.md

```diff
@@ -0,0 +1,596 @@
+
+By embedding the identity of the Node the Pod is running on, we can cross-reference this information with an identity
+passed along to the external service, thus removing the ability for a malicious actor to 'replay' a projected token
+from another Node.
+
+This will be implemented as an additional `node` entry in the private claims embedded into each JWT returned by the
+TokenRequest API, in a similar manner to how the ServiceAccount, Pod or Secret is referenced.
+
+Additionally, to provide a robust means of tracking token usage within the audit log we can embed a unique identifier for
+each token which is can then also be recorded in future audit entries made by this token.
+
+## Motivation
+
+### Goals
+
+* Embedding information about the Node that a pod is running on into signed JWTs.
+* Make it easier to track the actions a single token has taken, and cross-reference that back to the origin of the token
+  (via audit log inspection).
+* Provide a means of checking whether a Pod's token is associated with the same Node as it was associated with when the
+  initial TokenRequest was made (via TokenReview).
+
+### Non-Goals
+
+* Embedding requester information beyond just the requesting node. This is discussed further in the alternatives
+  considered section, and a future KEP may revisit this.
+* Embedding information beyond the Node name and UID into the token. We aim to mimic what is done with the ref fields
+  for secret, pod and serviceaccount (not introduce any additional properties).
+* Changing default behaviour of the TokenReview API to enforce the referenced Node object still exists.
+
+## Proposal
+
+### Embedding Pod's bound Node information in tokens
+
+The kube-apiserver will be extended to automatically embed the `name` and `uid` of the *Node* a Pod is associated
+with (via `spec.nodeName`) in generated tokens when a TokenRequest `create` call is serviced.
+
+As the 'pod' is already available in this area of code, which contains the `nodeName`, we will just need to plumb
+through a Getter for Node objects into the TokenRequest storage layer so the node's UID can be fetched, similar to
+what is done for pod & secret objects.
+
+### Extending TokenReview to allow cross-checking the embedded Node information with existing Node objects
+
+The TokenReview API will also be extended to check whether a JWT's embedded node information is still valid/current,
+and if not, will reject the review/indicate to the client that the token mismatches with the current state of Nodes.
+
+This will involve first checking whether the Node object with the name given in the JWT still exists, and if it does,
+validating whether the UID of that Node is equal to the UID embedded in the token.
+
+This means administrators can **delete node objects to invalidate all JWTs issued that embed that Node's info**.
+
+To avoid unexpected breaking changes in behaviour, this behaviour will not only be protected by the feature flag whilst
```

Comment:
Might want to move this up because until you get down to the end, the above sounds like a breaking change.